Appsilon/shiny.benchmark
# Make system functions silent when debug = FALSE

### Link to the Issue

close #28 

### Definition of Done

The user should be able to run the functions without receiving messages from system calls (unless debug = TRUE).

### How to test changes

Steps for testing:

- unzip file
- open app project under app
- from terminal run `bash ../setting_branches.sh` to create all the branches we are interested
- you may need to install shiny.performance manually (build source package and install it)
- run `bash ../run_tests.sh` to run all tests so far

[all_tests.zip](https://github.com/Appsilon/experimental.performance/files/9842008/all_tests.zip)

The messages in user's screen should be less frequent now.

### Tasks for PR author


### General policy:

- If applicable - add instructions for testing
- If there’s no issue, create it. Each issue needs to be well defined and described.
- All interaction with a user, user-facing messages, plots, reports etc. are written from the perspective of the person using or receiving it. They are understandable and helpful to this person. If a user sees an error message, there is a call to action, i.e. the user knows what to do to fix it.
- README, other documentation and code comments that we have is updated with all information related to the change.
- All code has been peer-reviewed before merging into any main branch
- All changes have been merged into the main branch we use for development.
- Continuous integration checks (linter, unit tests, integration tests) are configured and pass.
- Optional: unit tests added for all new or changed logic.
- All task requirements satisfied. If not describe it here. The reviewer is responsible to verify each aspect of the task.
- Change covers only things in task. Please create new PR if you want to fix something else.
